include/librealsense2/h/rs_option.h

```diff
@@ -340,6 +340,22 @@ extern "C" {
     */
     const char* rs2_get_option_value_description(const rs2_options* options, rs2_option option, float value, rs2_error ** error);
 
+    /**
+    * sets a callback in case an option in this options container value is updated
+    * \param[in] options    the options container
+    * \param[in] callback   callback function pointer to update on value changes
+    * \param[out] error     if non-null, receives any error that occurs during this call, otherwise, errors are ignored
+    */
+    void rs2_on_option_value_update( const rs2_options * options, rs2_option_value_update_callback_ptr callback, rs2_error ** error );
```

Comment:
The previous terminology we've used is `set` (not `on`) and `changed` (not `update`). E.g., `rs2_set_devices_changed_callback` and `rs2_playback_device_set_status_changed_callback`.

Maybe `rs2_set_option_changed_callback`?

Reply:
Agreed, from C API we need to support only a single callback. Removing can be done either by overriding the previous or releasing the sensor. Should be good enough.

Reply:
BTW, I'm revising the name: `rs2_set_options_changed_callback` (option->options)

Reply:
Done